main.go

```diff
@@ -50,9 +50,18 @@ func (mycli *MyClient) eventHandler(evt interface{}) {
 			return
 		}
 		// Read the response
-		buf := new(bytes.Buffer)
-		buf.ReadFrom(resp.Body)
-		newMsg := buf.String()
+		newMsg, err := ioutil.ReadAll(resp.Body)
+		resp.Body.Close()
```

Comment:
This should be deferred... no?

Reply:
No need to, this code closes it immediately after use. A defer will keep it open until the function returns or panics and then that's when the close will happen.